components/tidb_query/src/codec/mysql/time/mod.rs

```diff
@@ -98,12 +98,14 @@ fn ymd_hms_nanos<T: TimeZone>(
         })
 }
 
+// Safety: caller must ensure `bs` is valid utf8.
```

Comment:
The comment conflicts with the new implementation.

Reply:
Where is the new implementation? I don't see a conflict here

Reply:
ping @hicqu

Reply:
I mean if we mark the function as `unsafe`, the comment is not necessary any more, I guess.

Reply:
Ah, I see. I think that if the comment is safe, then the caller can assume that calling the function is safe for any inputs. If the function is unsafe, that means that the caller must ensure some invariant, the comment states what that invariant is.